src/emulator/storage/files.ts

```diff
@@ -221,8 +221,12 @@ export class StorageLayer {
     if (!upload) {
       return undefined;
     }
+
+    if (upload.status !== UploadStatus.FINISHED) {
```

Comment:
What happens if you try to cancel an already cancelled upload (ie. `upload.status === UploadStatus.CANCELLED`)? My guess is that this will crash due to trying to delete a non-existent file.

Please add a test case for this

Reply:
Done.